frontend/src/components/common/WeeklyEventItems.tsx

```diff
@@ -1,55 +1,86 @@
 import {
   Badge,
-  Box,
-  Button,
-  Center,
   Container,
-  Stack,
   Text,
+  useDisclosure,
+  VStack,
 } from "@chakra-ui/react";
-import React from "react";
+import React, { useEffect, useState } from "react";
 
-import { colorMap } from "../../constants/DaysInWeek";
+import DonorAPIClient from "../../APIClients/DonorAPIClient";
+import { colorMap, convertTime } from "../../constants/DaysInWeek";
+import { DonorResponse } from "../../types/DonorTypes";
+import { Schedule } from "../../types/SchedulingTypes";
+import WeeklyEventItemPopUp from "./WeeklyEventItemPopUp";
 
 type DefaultWeeklyEventItemProps = {
-  title: string;
+  schedule: Schedule;
   date: string;
-  frequency: string;
 };
 
 const DefaultWeeklyEventItem = ({
-  title,
+  schedule,
   date,
-  frequency,
 }: DefaultWeeklyEventItemProps) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [donor, setDonor] = useState<DonorResponse>();
+
+  useEffect(() => {
+    const getDonor = async () => {
+      const donorResponse = await DonorAPIClient.getDonorById(
+        schedule!.donorId,
+      );
+      setDonor(donorResponse);
+    };
+
+    getDonor();
+  }, []);
+
   return (
-    <Container
-      color="#FAFCFE"
-      borderWidth="1px"
-      borderRadius="8px"
-      borderColor="#D8DDE0"
-      alignItems="center"
-      centerContent
-      py="24px"
-      px="30px"
-    >
-      <Text textAlign="center" textStyle="desktopBodyBold" mb="8px">
-        {title}
-      </Text>
-      <Text textAlign="center" textStyle="desktopSmall" mb="16px">
-        {date}
-      </Text>
-      <Badge
-        color={`${(colorMap as any)[frequency]}.100`}
-        backgroundColor={`${(colorMap as any)[frequency]}.200`}
-        textStyle="desktopSmall"
-        py="6px"
-        ph="14px"
-      >
-        {" "}
-        {frequency}{" "}
-      </Badge>
-    </Container>
+    <>
+      {
+        donor ?
+          <>

```

Comment:
```suggestion

```
don't need this react fragment

Reply:
the react fragment is required since the jsx return requires at least one parent element. On removing it you can see that error show up.